internal/planner/mapper/mapper.go

```diff
@@ -569,6 +570,32 @@ func resolveAggregates(
 	return fields, nil
 }
 
+// fixAggregateNestedTargets ensures that selections of
+// JSON objects are not interpreted as joins.
+//
+// This can happen when an aggregate contains a filter
+// on a JSON object, but we can't tell if it is a relation
+// until the child mapping is created.
+func fixAggregateNestedTargets(
```

Comment:
suggestion: I got stuck on this function name a little too long for my liking lol. I think the `fix` prefix threw me off. This fonction does something that is very specific to JSON objects so my suggestion is to make that clearer in the name. What we are really doing here is filtering out/removing JSON sub-fields from the selection set.

Also, this is used at the moment specifically for aggregates but there is nothing in the function itself that handles aggregate specific objects.

As a result of the above, I would suggest renaming to `removeJSONSubFields`.

Reply:
Renamed to `removeJSONSubFields` as suggested